src/Jh/DataImportMagento/Writer/ShipmentWriter.php

```diff
@@ -45,25 +45,45 @@ public function writeItem(array $item)
         if (!isset($item['order_id'])) {
             throw new WriterException('order_id must be set');
         }
-
         $order = clone $this->orderModel;
         $order->loadByIncrementId($item['order_id']);
-
         if (!$order->getId()) {
             throw new WriterException(sprintf('Order with ID: "%s" cannot be found', $item['order_id']));
         }
 
+        $customerEmail = $order->getCustomerEmail();
+
         try {
             $shipment = $order->prepareShipment();
             $shipment->register();
             $shipment->getOrder()->setData('is_in_process', true);
 
+            if (isset($customerEmail)) {
+                $shipment->sendEmail(
+                    true,
+                    sprintf(
+                        'Your Order has been shipped by %s and your tracking number is %s',
+                        $item['carrier'],
+                        $item['tracking_number']
+                    )
+                );
+                $shipment->setEmailSent(true);
+            }
+
             $transactionSave = clone $this->transactionResourceModel;
             $transactionSave
                 ->addObject($shipment)
                 ->addObject($shipment->getOrder())
                 ->save();
 
+            \Mage::getModel('sales/order_shipment_track')
```

Comment:
Also, none of this data may not be set. Other projects using this writer do not set tracking information. So all of this should be wrapped around an `isset($item['tracks'])` or something similar